core/templates/dev/head/pages/exploration_editor/statistics_tab/LearnerActionRenderServiceSpec.js

```diff
@@ -0,0 +1,316 @@
+              value: 30
+            }
+          }, 1
+        ));
+        expect(displayBlocks[0][3]).toEqual(this.laof.createNew(
+          'ExplorationQuit', {
+            state_name: {
+              value: 'stateName1'
+            },
+            time_spent_in_state_secs: {
+              value: 120
+            }
+          }, 1
+        ));
+      });
+
+    it('should assign multiple learner actions at same state to same block.',
+      function() {
+        this.ps.recordExplorationStartAction('stateName1');
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordExplorationQuitAction('stateName1', 120);
+
+        var learnerActions = this.ps.getPlaythrough().actions;
+
+        var displayBlocks = this.lars.getDisplayBlocks(learnerActions);
+        expect(displayBlocks.length).toEqual(1);
+        expect(displayBlocks[0].length).toEqual(8);
+        expect(displayBlocks[0][0]).toEqual(this.laof.createNew(
+          'ExplorationStart', {
+            state_name: {
+              value: 'stateName1'
+            }
+          }, 1
+        ));
+        expect(displayBlocks[0][7]).toEqual(this.laof.createNew(
+          'ExplorationQuit', {
+            state_name: {
+              value: 'stateName1'
+            },
+            time_spent_in_state_secs: {
+              value: 120
+            }
+          }, 1
+        ));
+      });
+
+    it('should render tables for MultipleIncorrectSubmissions issue block.',
+      function() {
+        this.ps.recordExplorationStartAction('stateName1');
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordExplorationQuitAction('stateName1', 120);
+
+        var learnerActions = this.ps.getPlaythrough().actions;
+
```

Comment:
This newline and some others are a bit excessive. Try to only use them to separate each of the three "pieces" of a test:
 1. Prepare the thing to be tested.
 2. Do the thing to be tested.
 3. Check that expectations are met.

Reply:
Done.